ledger/catchpointtracker.go

```diff
@@ -364,6 +379,10 @@ func (ct *catchpointTracker) newBlock(blk bookkeeping.Block, delta ledgercore.St
 		}
 		ct.accountDataResourceSeparationRound = blk.BlockHeader.Round + basics.Round(catchpointLookback)
 	}
+
+	if config.Consensus[blk.CurrentProtocol].EnableOnlineAccountCatchpoints {
```

Comment:
this will get executed every call to newBlock, whereas above there is a check for ct.accountDataResourceSeparationRound == 0. Should we keep the same logic as used for accountDataResourceSeparationRound?

Also, why do we need to use an atomic int? and why don't we have to do all the same round offset logic that is currently used to handle accountDataResourceSeparationRoudn in `calculateFirstStageRounds()` and `produceCommittingTask()`?